### ISSUE TITLE
Is DNSCrypt for Mac completely dead?

### ISSUE THREAD

**Issue by GitTheHellOutaHere** [alterstep/dnscrypt-osxclient#165]
For the past couple weeks or so the app hasn’t worked at all for me.
Whenever I connect to ANY server -  **Yes, I mean ANY - anywhere around the globe** - my Internet connection is dropped.

Disabling DNSCrpt restores my connection immediately.

Although Network Diagnostic shows red for Internet and Server when DNSCrypt is enabled, the app still thinks it’s connected and displays the padlock.

I tried removing all the IP addresses I’d entered in the Advanced tab just in case it made any difference, but it didn’t.

I re-installed over the existing version in case something got deleted or corrupted but nothing changed so I uninstalled it altogether.

Even if the project was dead, why would the app suddenly stop working?
Are any servers still working for others on Mac?

OS X 10.12.6
DNSCrypt 1.0.14

**Comment by JayBrown** [comment]
Everything's fine here (10.11.6). I had one server drop out on me, so I had to switch back to my original choice, but the app itself is fine. It has some hickups, e.g. when waking from sleep: then it sometimes seems to connect twice to the DNS, but afterwards it's fine. Maybe there are other hickups or bugs, don't know, but anytime I've lost my internet connection,clearing the DNS cache, renewing DHCP and a reboot have always helped.

At any rate, there's always the original DNSCrypt, which definitely works for macOS (already at v2.0): https://github.com/jedisct1/dnscrypt-proxy

**Comment by GitTheHellOutaHere** [comment]
Yeah, I posted this issue there first and was told _"DNSCrypt-OSXClient (or whatever it is called) is another project. It's a user interface that needs an update or a rewrite."_

That’s just a repository, not an active project:
**"DNSCrypt-Proxy repository, frankly maintained for what it does (no new features planned)"**

One server down is nothing - I’ve had several servers that had been working fine suddenly stop working. Most were temporary and began working again at some point. But to have every one become unusable at once was really unexpected.

I reboot and clean caches/logs regularly, even the reinstall didn’t help.

**Comment by GitTheHellOutaHere** [comment]
I just enabled DNSCrypt Menubar to give it another shot and now DNSCrypt is working.

I toyed with the Xfinity firewall settings and changed my wifi signal’s name on their website the other day, but the computer connection is Ethernet so the wifi shouldn’t have mattered.

I think that’s the first time I’ve touched the firewall and I increased the security level so I don’t believe it was causing the blocked connections.

**Comment by evilvibes** [comment]
I think you have a miss understanding dnscrypt-osxclient uses dnscrypt-proxy 1.9.4 it has not been updated to 1.9.5 and there will be no more updates for dnscrypt-proxy v1 as it just being maintained by @Dyne (https://github.com/dyne/dnscrypt-proxy). dnscrypt-osxclient has not been updated in a longtime.

@jedisct1 has now released dnscrypt-proxy v2 that is being maintained and updated. (https://github.com/jedisct1/dnscrypt-proxy) 

How to install it on Mac is located here (https://github.com/jedisct1/dnscrypt-proxy/wiki/installation#installing-dnscrypt-proxy-on-macos-as-a-system-service)

Hope this helps

**Comment by GitTheHellOutaHere** [comment]
No one mentioned v1.9.4 or v1.9.5.
And dnscrypt-proxy is now up to v2.0.1.

I thought dnscrypt-osxclient _was_ the Mac version, I hadn’t realized that other project also has a Mac version. So this project is dead but the other is the newer (improved?) replacement?

It runs as a "system service," so there’s no longer a menu item since it’s not an app?
If so, how are different servers selected?

How is it better than what I’m using now?

Is there really any reason to use DNSCrypt at all, at least for US users?
I can see if someone’s in a restricted country it may help, but as for privacy, doesn’t it do the same as me searching for DNS server addresses and entering them manually in Network Preferences? 


The install instructions for dnscrypt-proxy are easy enough, though I prefer things that come with installers.

**Comment by evilvibes** [comment]
`"No one mentioned v1.9.4 or v1.9.5.
And dnscrypt-proxy is now up to v2.0.1."`

`"I thought dnscrypt-osxclient was the Mac version, I hadn’t realized that other project also has a Mac version. So this project is dead but the other is the newer (improved?) replacement?"`

This mac client is for dnscrypt-proxy v1 that's the point for it to make it easier for people to use the dnscrypt-proxy. 

`"It runs as a "system service," so there’s no longer a menu item since it’s not an app?
If so, how are different servers selected?"`

Within the [dnscrypt-proxy.toml](https://github.com/jedisct1/dnscrypt-proxy/blob/master/dnscrypt-proxy/example-dnscrypt-proxy.toml) file.

`"How is it better than what I’m using now?"`

Told you here "Also here is the Differences from v1 to v2 (https://github.com/jedisct1/dnscrypt-proxy/wiki/Differences-to-v1)"

`"Is there really any reason to use DNSCrypt at all, at least for US users?"`

That's up to you.

Goodluck :)

**Comment by chrisloftus** [comment]
So this app is dead? 😢

**Comment by JayBrown** [comment]
Nobody knows. :)

**Comment by chrisloftus** [comment]
Doesn't seem to connect to any resolvers anymore.

**Comment by jedisct1** [comment]
A simple menubar app just to change the resolver to 127.0.0.1 and back to DHCP would be useful.

**Comment by JayBrown** [comment]
There's the *[DNS Switcher](https://getbitbar.com/plugins/Network/dnsswitcher.1d.sh)* shell script for **[BitBar](https://github.com/matryer/bitbar)**. The code is [here](https://raw.githubusercontent.com/matryer/bitbar-plugins/master/Network/dnsswitcher.1d.sh), and I guess you can then add some non-DNScrypt servers, and of course 127.0.0.1 (or rather: 127.0.0.54) for DNScrypt.

Come to think of it, it's surely possible to create a BitBar script (in bash, JavaScript, Swift etc.) that mimics the functionality of the DNScrypt OS X client, i.e. gets the current list of servers, presents them in the menu, changes the toml if you select one, with additional functionality to disable/enable DNScrypt, tweak some other other config data.

You could even use the the same icons as the current v1 OS X client; just integrate them as base64 in the script. :)

**Comment by GitTheHellOutaHere** [comment]
Well golly, that all sounds simple, Jay.

Why don’t you go on ahead and compile that (or whatever needs to be done to accomplish what you’ve suggested), then report back.

I’ll be waiting.

TIA!

**Comment by jedisct1** [comment]
Oh, I didn't remember about BitBar. This is great!

**Comment by jedisct1** [comment]
Just whipped up a BitBar plugin to control dnscrypt-proxy usage:

https://github.com/jedisct1/bitbar-dnscrypt-proxy-switcher/

It does everything I needed from dnscrypt-osxclient :)

**Comment by JayBrown** [comment]
Works great. Created two PRs, one to use the original alterstep/stammy icons, the other to account for the different commands per macOS version when clearing the DNS cache.

**Comment by JayBrown** [comment]
Scripted my own BitBar plugin, derived from Frank's original, but with some additional functionality & infos: https://github.com/JayBrown/DNSCrypt-Menu
